src/core/bsequence.ml

```diff
@@ -0,0 +1,49 @@
+
+type +'a node =
+    | Nil
+    | Cons of 'a * 'a t
+and 'a t = unit -> 'a node
+
+let empty () = Nil
+
+let return x () = Cons(x, empty)
```

Comment:
Good for who? Should we optimize for the few that have already demonstrated a great amount of perseverance by learning category theory, or the many who have not?

As an example, to call an external function that accepts a nullable value, e.g `external f : int Js.null -> unit = ..." you currently have to write (and read) `f (Js.Null.return 0)`. That's not a great experience and severely undermines what's otherwise a damn good FFI system. Most also seem to get along just fine with the option monad without needing a `return` function.

That's not to say it wouldn't be a good to have a standard monad "interface"/module signature, but I don't see great benefit in that interface adopting standard terminology, compared to the benefit of avoiding incomprehensible terminology.